### PR TITLE
New App Layout FABs

### PR DESCRIPTION
**Pull request by ericdecanini**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Adds the new FABs to be used in the New App Layout. These will be hidden behind the New App Layout feature flag

## Motivation and context

Closes https://github.com/vector-im/element-android/issues/6582

## Screenshots / GIFs

| | Feature Flag Off | Feature Flag On |
|-|-|-|
| Light |  |  |
| Dark |  |  |
| Black |  |  |

## Tests

- Enable the New App Layout feature flag
- Sign in and see the FABs

Currently the FABs do not do anything on click. This will be handled in a separate PR

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Comment by ericdecanini** [comment]
LGTM

**Review by amshakal — APPROVED** [review]
Thank you!

We might change the icon for space switching based on feedback we recieve this week but the rest is perfect!

File: vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt [hunk 1]
```diff
@@ -108,6 +110,41 @@ class HomeRoomListFragment @Inject constructor(
...
 
+    private fun setupFabs() {
+        showFABs()
+
+        views.newLayoutCreateChatButton.setOnClickListener {
+            // Click action for create chat modal goes here
+        }
+
+        views.newLayoutOpenSpacesButton.setOnClickListener {
+            // Click action for open spaces modal goes here
```
Comment (fedrunov) [inline vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt:121]: worth to put here a todo with issue number

File: vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt [hunk 2]
```diff
@@ -108,6 +110,41 @@ class HomeRoomListFragment @Inject constructor(
         views.roomListView.adapter = concatAdapter
     }
 
+    private fun setupFabs() {
+        showFABs()
+
+        views.newLayoutCreateChatButton.setOnClickListener {
+            // Click action for create chat modal goes here
```
Comment (fedrunov) [inline vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt:117]: missing implementation

**Review by fedrunov — CHANGES_REQUESTED** [review]

**Review by fedrunov — APPROVED** [review]

**Pull request merged**